src/renderers/shared/fiber/ReactFiberCompleteWork.js

```diff
@@ -200,8 +201,12 @@ module.exports = function<T, P, I, TI, PI, C, CX>(
           // Even better would be if children weren't special cased at all tho.
           const instance : I = workInProgress.stateNode;
           const currentHostContext = getHostContext();
-          if (prepareUpdate(instance, type, oldProps, newProps, currentHostContext)) {
-            // This returns true if there was something to update.
+          const updatePayload = prepareUpdate(instance, type, oldProps, newProps, rootContainerInstance, currentHostContext);
+          // TODO: Type this specific to this type of component.
+          workInProgress.updateQueue = (updatePayload : any);
+          // If the update payload indicates that there is a change or if there
+          // is a new ref we mark this as an update.
+          if (updatePayload || current.ref !== workInProgress.ref) {
```

Comment:
Refs shouldn't schedule an `Update` effect anymore, since we added the `Ref` effect